Parse X-Forwarded-Prefix request header

chemicL opened the pull request:
The X-Forwarded-Prefix can be obtained via `HttpServerRequest#forwardedPrefix()`.

chemicL commented on reactor-netty-http/src/main/java/reactor/netty/http/server/ConnectionInfo.java:
```diff
@@ -94,12 +96,18 @@ static ConnectionInfo from(Channel channel, HttpRequest request, boolean secured
 
 	ConnectionInfo(SocketAddress hostAddress, String hostName, int hostPort,
 			SocketAddress remoteAddress, String scheme, boolean isInetAddress) {
+		this(hostAddress, hostName, hostPort, remoteAddress, scheme, isInetAddress, "");
```
I wonder if we'd actually like to provide the means to tell the difference between a missing header and the content being empty. If so, this could be actually `null` instead of empty by default. WDYT @violetagg ?

violetagg replied:
yeah may be we should return `null` if there is no header or `forwarded=false`

chemicL commented on reactor-netty-http/src/main/java/reactor/netty/http/server/ConnectionInfo.java:
```diff
@@ -94,12 +96,18 @@ static ConnectionInfo from(Channel channel, HttpRequest request, boolean secured
+		this(hostAddress, hostName, hostPort, remoteAddress, scheme, isInetAddress, "");
+	}
+
+	ConnectionInfo(SocketAddress hostAddress, String hostName, int hostPort,
```
I added a new constructor, but I can absolutely go ahead and modify all the usages of the above constructor to add the last parameter and remove the above.

violetagg replied:
I like it with a new constructor

violetagg reviewed (approved):
nice work

chemicL commented:
@violetagg I addressed the aspect of distinguishing missing header vs empty content and also added the handling of comma-separated components with removal of the trailing forward slashes in each component.

chemicL commented:
@violetagg as discussed, I updated the PR to include validation whether the header value starts with a slash.

violetagg commented on reactor-netty-http/src/main/java/reactor/netty/http/server/DefaultHttpForwardedHeaderHandler.java:
```diff
@@ -138,7 +138,11 @@ private static String parseForwardedPrefix(String prefixHeader) {
 		}
-		return prefix.toString();
+		String parsedPrefix = prefix.toString();
+		if (!parsedPrefix.isEmpty() && DEFAULT_FORWARDED_HEADER_VALIDATION && !parsedPrefix.startsWith("/")) {
```
instead of `startsWith("/")` what about `tempUri.charAt(0) != '/'`

chemicL replied:
Yep, that's better. Applied the change.
